src/filesystem/api.h

```diff
@@ -198,6 +198,15 @@ Status MakeDirectory(const std::string& dir, const bool recursive);
 /// \return Error status
 Status MakeTemporaryDirectory(const FileSystemType type, std::string* temp_dir);
 
+/// Create a temporary directory of the specified filesystem type
+/// in a provided location.
+/// \param type The type of the filesystem.
+/// \param  dir_path The specified path.
+/// \param temp_dir Returns the path to the temporary directory.
+/// \return Error status
+Status MakeTemporaryDirectory(
+    const FileSystemType type, std::string dir_path, std::string* temp_dir);
```

Comment:
A small thing, I think the `dir_path` can be declared as `const std::string& dir_path` here, and leave the `LocalFileSystem::MakeTemporaryDirectory(std::string dir_path, ...)` as it is. Then, we can save a copy of `dir_path` when passing the variable from the caller to `MakeTemporaryDirectory()` here, and only copy it once when passing `dir_path` from here to `LocalFileSystem::MakeTemporaryDirectory()`.

Reply:
I'll make sure to check and introduce this in a follow-up pr with the bug fix.   Thanks for suggestion!